module/VuFindSearch/src/VuFindSearch/Backend/Blender/Response/Json/RecordCollection.php

```diff
@@ -0,0 +1,441 @@
+                $record->setSourceIdentifiers(
+                    $record->getSourceIdentifier(),
+                    $backendId
+                );
+                if ($label) {
+                    $record->addLabel($label, 'source');
+                }
+                $backendRecords[$backendId][] = $record;
+            }
+
+            foreach ($collection->getErrors() as $error) {
+                if (is_string($error) && $label) {
+                    $error = [
+                        'msg' => '%%error%% -- %%label%%',
+                        'tokens' => [
+                            '%%error%%' => $error,
+                            '%%label%%' => $label
+                        ],
+                        'translate' => true,
+                        'translateTokens' => true,
+                    ];
+                }
+                $this->addError($error);
+            }
+        }
+
+        $records = [];
+        $backendIds = array_keys($backendRecords);
+        // Filter out unavailable backends from initial results source list:
+        $initialResults = array_values(
+            array_filter(
+                $this->initialResults,
+                function ($s) use ($backendIds) {
+                    return in_array($s, $backendIds);
+                }
+            )
+        );
+        for ($pos = 0; $pos < $limit; $pos++) {
+            $backendId = $this->getBackendAtPosition(
+                $pos,
+                $blockSize,
+                $backendIds,
+                $initialResults
+            );
+            if (!empty($backendRecords[$backendId])) {
+                $this->add(array_shift($backendRecords[$backendId]), false);
+            }
+        }
+
+        $this->response['facet_counts']['facet_fields']
+            = $this->getMergedFacets($collections);
+
+        return $backendRecords;
+    }
+
+    /**
+     * Slice the record collection
+     *
+     * @param int $offset Offset
+     * @param int $limit  Limit
+     *
+     * @return void
+     */
+    public function slice(int $offset, int $limit): void
+    {
+        $this->records = array_slice(
+            $this->records,
+            $offset,
+            $limit
+        );
+    }
+
+    /**
+     * Add an error message
+     *
+     * @param mixed $error Error
+     *
+     * @return void
+     */
+    public function addError($error): void
+    {
+        if (!in_array($error, $this->errors)) {
+            $this->errors[] = $error;
+        }
+    }
+
+    /**
+     * Return any errors.
+     *
+     * Each error can be a translatable string or an array that the Flashmessages
+     * view helper understands.
+     *
+     * @return array
+     */
+    public function getErrors()
+    {
+        return $this->errors;
+    }
+
+    /**
+     * Set the source backend identifier.
+     *
+     * @param string $identifier Backend identifier
+     *
+     * @return void
+     */
+    public function setSourceIdentifier($identifier)
```

Comment:
This method is already defined in VuFindSearch\Response\AbstractRecordCollection. Is there a reason to redefine it here?

Reply:
I added a comment about the reason.